prow/cmd/clonerefs/main.go

```diff
@@ -30,6 +31,39 @@ import (
 	"k8s.io/test-infra/prow/pod-utils/clone"
 )
 
+type options struct {
+	srcRoot string
+	log     string
+
+	gitUserName  string
+	gitUserEmail string
+
+	refs gitRefs
+}
+
+func (o *options) Validate() error {
+	if o.srcRoot == "" {
+		return errors.New("no source root specified")
+	}
+
+	if o.log == "" {
+		return errors.New("no log file specified")
+	}
+
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	flag.StringVar(&o.srcRoot, "src-root", "", "Where to root source checkouts")
```

Comment:
should this use a flagset? :shrug:

Reply:
If so, this should take in args and return an error which we can handle gracefully / log instead of the internal flag panic

Reply:
discussed on slack, will punt to later :-)